File: leetcode/strings-and-array/cpp/1431.kids-with-the-greatest-number-of-candies.cpp
/*
 * @lc app=leetcode id=1431 lang=cpp
 *
 * [1431] Kids With the Greatest Number of Candies
 */

// @lc code=start
#include <algorithm>

class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> res;
        int max = *max_element(candies.begin(), candies.end());
        for (auto kid : candies) {
            if (kid + extraCandies >= max) res.push_back(true);
            else res.push_back(false);
        }
        return res;
    }
};
// @lc code=end

File: leetcode/strings-and-array/py/1431.kids-with-the-greatest-number-of-candies.py
#
# @lc app=leetcode id=1431 lang=python3
#
# [1431] Kids With the Greatest Number of Candies
#

# @lc code=start
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        maxNum = max(candies)
        res = [] # declare + initialize a list
        for n in candies:        # remember the goddamn colons
            if n + extraCandies >= maxNum:
                res.append(True)
            else:
                res.append(False)
        return res
# @lc code=end

